addons/hr/static/src/models/public_employee/public_employee.js

```diff
@@ -0,0 +1,205 @@
+                } else {
+                    const partnerNameGet = data['user_partner_id'];
+                    const partnerData = {
+                        display_name: partnerNameGet[1],
+                        id: partnerNameGet[0],
+                    };
+                    const userNameGet = data['user_id'];
+                    const userData = {
+                        id: userNameGet[0],
+                        partner: [['insert', partnerData]],
+                        partnerDisplayName: userNameGet[1],
+                    };
+                    data2.user = [['insert', userData]];
+                }
+            }
+            return data2;
+        }
+
+        /**
+         * Performs the `read` RPC on the `hr.employee.public`.
+         *
+         * @static
+         * @param {Object} param0
+         * @param {Object} param0.context
+         * @param {string[]} param0.fields
+         * @param {integer[]} param0.ids
+         */
+        static async performRpcRead({ context, fields, ids }) {
+            const employeesData = await this.env.services.rpc({
+                model: 'hr.employee.public',
+                method: 'read',
+                args: [ids],
+                kwargs: {
+                    context,
+                    fields,
+                },
+            });
+            this.env.models['hr.employee'].insert(employeesData.map(employeeData =>
+                this.env.models['hr.employee'].convertData(employeeData)
+            ));
+        }
+
+        /**
+         * Performs the `search_read` RPC on `hr.employee.public`.
+         *
+         * @static
+         * @param {Object} param0
+         * @param {Object} param0.context
+         * @param {Array[]} param0.domain
+         * @param {string[]} param0.fields
+         */
+        static async performRpcSearchRead({ context, domain, fields }) {
+            const employeesData = await this.env.services.rpc({
+                model: 'hr.employee.public',
+                method: 'search_read',
+                kwargs: {
+                    context,
+                    domain,
+                    fields,
+                },
+            });
+            this.env.models['hr.employee'].insert(employeesData.map(employeeData =>
+                this.env.models['hr.employee'].convertData(employeeData)
+            ));
+        }
+
+        /**
+         * Checks whether this employee has a related user and partner and links
+         * them if applicable.
+         */
+        async checkIsUser() {
+            return this.env.models['hr.employee'].performRpcRead({
+                ids: [this.id],
+                fields: ['user_id', 'user_partner_id'],
+                context: { active_test: false },
+            });
+        }
+
+        /**
+         * Gets the canonical chat between the user of this employee and the
+         * current user.
+         *
+         * If a chat is not appropriate a notification is displayed instead.
```

Comment:
missing comma

Reply:
I don't think it's grammatically required but I don't mind adding it.